commons/src/main/java/com/powsybl/commons/compress/SafeZipInputStream.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.commons.compress;
+
+import java.io.IOException;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import com.powsybl.commons.io.ForwardingInputStream;
+
+public class SafeZipInputStream extends ForwardingInputStream<ZipInputStream> {
+
+    private long bytesRead;
+    private final long maxBytesToRead;
+
+    public SafeZipInputStream(ZipInputStream in, long maxBytesToRead) {
+        super(in);
+        this.maxBytesToRead = maxBytesToRead;
+    }
+
+    public ZipEntry getNextEntry() throws IOException {
+        return this.getDelegate().getNextEntry();
```

Comment:
I think we could also limit the number of calls to `getNextEntry()` on the delegate, just in case this class is used later for more complex zips:
- new constructor parameter `int maxEntriesToRead`
- control in `getNextEntry()`: if counter > maxEntriesToRead then throw an IOException.